src/main/scala/mesosphere/marathon/upgrade/TaskReplaceActor.scala

```diff
@@ -36,11 +44,34 @@ class TaskReplaceActor(
   val periodicalRetryKills: Cancellable = context.system.scheduler.schedule(15.seconds, 15.seconds, self, RetryKills)
 
   override def preStart(): Unit = {
+    // TODO there is a race condition if this come before or after the subscribe. If it goes before
+    // TODO   the subscribe then we miss updates during that time. If it comes after, updates could
+    // TODO   still be received out of sync with this request for status.
+    if (app.healthChecks.nonEmpty) {
+      // TODO what should the timeout be for this?
+      val health = Await.result(healthCheckManager.statuses(appId), 5.seconds)
```

Comment:
Even though we use it elsewhere, I do not like the `Await` approach much. Better use non-blocking constructs like multiple receives in different states.